torchok/metrics/index_base_metric.py

```diff
@@ -160,35 +160,74 @@ def compute(self) -> float:
 
         if self.dataset_type == DatasetType.CLASSIFICATION:
             # if classification dataset
-            targets = torch.cat(self.targets).numpy()
+            group_labels = torch.cat(self.group_labels).numpy()
             # prepare data
-            relevant_idxs, gallery_idxs, query_row_idxs, query_as_relevant = self.prepare_classification_data(targets)
+            relevant_idxs, faiss_vector_idxs, \
+            query_row_idxs, query_as_relevant = self.prepare_classification_data(group_labels)
             # mock scores and query column indexes because it belongs to representation data
             scores = None
             query_column_idxs = None
         else:
             # if representation dataset
             scores = torch.cat(self.scores).numpy()
             query_idxs = torch.cat(self.query_idxs).numpy()
+            group_labels = torch.cat(self.group_labels).numpy()
             # prepare data
-            relevant_idxs, gallery_idxs, query_column_idxs,\
-                query_row_idxs, query_as_relevant = self.prepare_representation_data(query_idxs, scores)
+            relevant_idxs, faiss_vector_idxs, query_column_idxs, \
+            query_row_idxs, query_as_relevant = self.prepare_representation_data(query_idxs, scores)
 
         # build index
         vectors = vectors.astype(np.float32)
-        index = self.build_index(vectors[gallery_idxs])
+        index = self.build_index(vectors[faiss_vector_idxs])
 
-        # create relevant, closest generator
-        generator = self.query_generator(index, vectors, relevant_idxs, query_row_idxs,
-                                         gallery_idxs, query_as_relevant, self.search_k, scores, query_column_idxs)
+        # split query by group_label if metric compute target averaging
+        if self.group_averaging:
+            uniq_group_labels = np.unique(group_labels)
+            group_indexes_split = np.array([np.where(group_labels == uniq_group_labels[i])[0]
```

Comment:
`group_indexes_split = np.array([np.where(group_labels == label)[0] for label in uniq_group_labels])`